laurat007/assignment2/AncestorsTest.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package assignment_2;
+
+import java.util.ArrayList;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import junit.framework.TestCase;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+/**
+ *
+ * @author Laura
+ */
+public class AncestorsTest extends TestCase {
+
+    public AncestorsTest() {
+    }
+
+    Ancestors.BinaryTreeNode<Integer> buildTree1() {
+        // build a binary tree
+        Ancestors.BinaryTreeNode<Integer> tree = new Ancestors.BinaryTreeNode<>(7);
+        tree.addLeft(3);
+        tree.addRight(4);
+        tree.addLeft(2);
+        tree.addBalanced(5, 0);
+        tree.addBalanced(8, 1);
+        tree.addBalanced(1, 1);
+        tree.addBalanced(6, 1);
+        return tree;
+    }
+
+    Ancestors.BinaryTreeNode<Integer> buildTree2() {
+        // build a binary tree
+        Ancestors.BinaryTreeNode<Integer> tree = new Ancestors.BinaryTreeNode<>(7);
+        return tree;
+    }
+
+    Ancestors.BinaryTreeNode<Integer> buildTree3() {
+        // build a binary tree
+        Ancestors.BinaryTreeNode<Integer> tree = new Ancestors.BinaryTreeNode<>(7);
+        tree.addLeft(3);
+        tree.addLeft(8);
+        tree.addLeft(10);
+        tree.addLeft(5);
+        tree.addLeft(4);
+        tree.addLeft(1);
+        return tree;
+    }
+
+    @Before
+    public void setUp() {
+        try {
+            super.setUp();
+        } catch (Exception ex) {
+            Logger.getLogger(AncestorsTest.class.getName()).log(Level.SEVERE, null, ex);
+        }
+    }
+
+    @After
+    public void tearDown() {
+        try {
+            super.tearDown();
+        } catch (Exception ex) {
+            Logger.getLogger(AncestorsTest.class.getName()).log(Level.SEVERE, null, ex);
+        }
+    }
+
+    // test getAncestors
+    @Test
+    public void Test1() {
```

Comment:
Try naming your tests in a more descriptive way. There are many conventions, for example you can have the name tell what it tests, what are special conditions and the expected outcome, for example: getAncestors_invalidNodes_returnsNull() etc.
Reading this method name you immediately can figure out what it tests and if you see this test failing you have an easy way to figure out what you might have broken with your changes. It benefits everyone :)

Reply:
Hope it's better now.